src/plugins/gs-plugin-eos.c

```diff
@@ -362,3 +362,30 @@ gs_plugin_remove_shortcut (GsPlugin	*plugin,
 	gs_app_remove_quirk (app, AS_APP_QUIRK_HAS_SHORTCUT);
 	return remove_app_from_shell (plugin, app, cancellable, error);
 }
+
+static gboolean
+app_is_flatpak (GsApp *app)
+{
+	const gchar *id = gs_app_get_id (app);
+	return id && (g_str_has_prefix (id, GS_FLATPAK_USER_PREFIX ":") ||
+		      g_str_has_prefix (id, GS_FLATPAK_SYSTEM_PREFIX ":"));
+}
+
+gboolean
+gs_plugin_app_install (GsPlugin *plugin,
+		       GsApp *app,
+		       GCancellable *cancellable,
+		       GError **error)
+{
+	if (!app_is_flatpak (app))
+		return TRUE;
+
+	/* We're only interested in already installed flatpak apps so we can
+	 * add them to the desktop */
```

Comment:
Yes. *gs_plugin_app_install* is called for every app that the user presses "Install". It's up for the plugins to do what they need in this method (usually plugins check if the app is "theirs" using the management plugin property before proceeding). I took advantage of this method in order to complete the installation on EOS (i.e. adding the icon to the desktop). This works because the plugin runs **after** the flatpak ones and I check if the app is **a flatpak** + **installed**.